src/DIRAC/Core/Utilities/Subprocess.py

```diff
@@ -269,40 +269,20 @@ def __poll(self, pid):
     def killChild(self, recursive=True):
         """kill child process
 
-        FIXME: this can easily be rewritten just using this recipe:
-        https://psutil.readthedocs.io/en/latest/index.html#kill-process-tree
-
         :param boolean recursive: flag to kill all descendants
         """
-        if self.childPID < 1:
-            self.log.error("Could not kill child", "Child PID is %s" % self.childPID)
-            return -1
-        os.kill(self.childPID, signal.SIGSTOP)
-        if recursive:
-            for gcpid in getChildrenPIDs(self.childPID, lambda cpid: os.kill(cpid, signal.SIGSTOP)):
-                try:
-                    os.kill(gcpid, signal.SIGKILL)
-                    self.__poll(gcpid)
-                except Exception:
-                    pass
-        self.__killPid(self.childPID)
-
-        # HACK to avoid python bug
-        # self.child.wait()
-        exitStatus = self.__poll(self.childPID)
-        i = 0
-        while exitStatus is None and i < 1000:
-            i += 1
-            time.sleep(0.000001)
-            exitStatus = self.__poll(self.childPID)
-        try:
-            exitStatus = os.waitpid(self.childPID, 0)
-        except os.error:
-            pass
-        self.childKilled = True
-        if exitStatus is None:
-            return exitStatus
-        return exitStatus[1]
+
+        parent = psutil.Process(self.childPID)
+        children = parent.children(recursive=recursive)
+        children.append(parent)
+        for p in children:
+            try:
+                p.send_signal(signal.SIGTERM)
+            except psutil.NoSuchProcess:
+                pass
+        _gone, alive = psutil.wait_procs(children)
```

Comment:
Do you want to use a timeout here?

Reply:
...maybe. I don't know, how long?

Reply:
Assuming there are no places that Subprocess is used where killing is inappropriate, 10 seconds?

For `Watchdog.py` I think it's definitely fine to have 10 seconds or less.